packages/core/core-flows/src/inventory/workflows/create-inventory-items.ts

```diff
@@ -3,23 +3,86 @@ import {
   createWorkflow,
   transform,
 } from "@medusajs/workflows-sdk"
-import {
-  attachInventoryItemToVariants,
-  createInventoryItemsStep,
-  validateInventoryItemsForCreate,
-} from "../steps"
+import { createInventoryItemsStep } from "../steps"
 
 import { InventoryNext } from "@medusajs/types"
+import { createInventoryLevelsWorkflow } from "./create-inventory-levels"
 
+type LocationLevelWithoutInventory = Omit<
+  InventoryNext.CreateInventoryLevelInput,
+  "inventory_item_id"
+>
 interface WorkflowInput {
-  items: InventoryNext.CreateInventoryItemInput[]
+  items: (InventoryNext.CreateInventoryItemInput & {
+    location_levels?: LocationLevelWithoutInventory[]
+  })[]
+}
+
+const buildLocationLevelMapAndItemData = (data: WorkflowInput) => {
+  const inventoryItems: InventoryNext.CreateInventoryItemInput[] = []
+  // Keep an index to location levels mapping to inject the created inventory item
+  // id into the location levels workflow input
+  const locationLevelMap: Record<number, LocationLevelWithoutInventory[]> = {}
+  let index = 0
+
+  for (const { location_levels, ...inventoryItem } of data.items || []) {
```

Comment:
Should we do `data.items.forEach` so you don't have to keep track of the index manually?